server/src/main/java/io/crate/analyze/relations/FullQualifiedNameFieldProvider.java

```diff
@@ -97,9 +100,23 @@ public Symbol resolveField(QualifiedName qualifiedName, @Nullable List<String> p
             tableNameMatched = true;
 
             AnalyzedRelation sourceRelation = entry.getValue();
-            Symbol newField = sourceRelation.getField(columnIdent, operation);
+            Symbol newField;
+            try {
+                newField = sourceRelation.getField(columnIdent, operation, errorOnUnknownObjectKey);
+            } catch (UnknownObjectKeyExceptionalControlFlow e) {
+                assert errorOnUnknownObjectKey == false : "errorOnUnknownObjectKey should not be true";
+                newField = new AliasSymbol(columnIdent.toString(), Literal.NULL);
```

Comment:
```suggestion
                newField = new AliasSymbol(columnIdent.sqlFqn(), Literal.NULL);
```

Would make it easier to see if this is the subscript notation or the dotted notation.